driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedGridFSHelper.java

```diff
@@ -32,19 +33,48 @@
 
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import static java.util.Objects.requireNonNull;
 
-final class UnifiedGridFSHelper {
+final class UnifiedGridFSHelper extends UnifiedHelper{
     private final Entities entities;
 
     UnifiedGridFSHelper(final Entities entities) {
         this.entities = entities;
     }
 
+    public OperationResult executeFind(final BsonDocument operation) {
+        GridFSFindIterable iterable = createGridFSFindIterable(operation);
+        try {
+             iterable.into(new ArrayList<>());
+            return OperationResult.NONE; // we don't expect results in the tests.
```

Comment:
Is this a future gotcha? Should we return OperationResult.NONE if the array is empty?

Reply:
Thank you for highlighting this! I've modified the code to return `OperationResult.NONE` when the array is empty. For all other cases, I've added an `UnsupportedOperationException` to prevent future issues. Currently, in our unified`GridFS` tests for the find operation, the `expectResult` format isn't specified.